azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -373,10 +373,23 @@ def _on_mqtt_disconnected(self, cause=None):
         else:
             logger.info("{}: _on_mqtt_disconnect called".format(self.name))
 
-        # Send an event to tell other pipeilne stages that we're disconnected. Do this before
+        # Send an event to tell other pipeline stages that we're disconnected. Do this before
         # we do anything else (in case upper stages have any "are we connected" logic.)
         self.send_event_up(pipeline_events_base.DisconnectedEvent())
 
+        # If there is no connection retry, cancel any transport operations waiting on response
+        # so that they do not get stuck there.
+        if not self.pipeline_root.pipeline_configuration.connection_retry:
+            logger.debug(
+                "{}: Connection Retry disabled - cancelling in-flight operations".format(self.name)
+            )
+            # TODO: Remove private access to the op manager (this layer shouldn't know about it)
```

Comment:
This is for a future PR to sort out. I don't like leaving a stopgap in here, but it's better than introducing a (temporary) API to get around it.